src/hooks/useTokenLockStates.ts

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from 'react';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { type Token, useGetColonyTokensWithLockedStatesQuery } from '~gql';
+
+const useTokenLockStates = (): Record<string, boolean> => {
+  const {
+    colony: { name },
+  } = useColonyContext();
+  const { data } = useGetColonyTokensWithLockedStatesQuery({
```

Comment:
as discussed on Discord, it may be a bit more granular and performant to loop over all the tokens of a given colony, call `getColoniesByNativeTokenId` for each token in a colony and smash that into a state variable.

Reply:
This is the way!